src/main/java/teammates/ui/controller/AdminEmailComposeSendAction.java

```diff
@@ -0,0 +1,295 @@
+        if(groupModeOn){     
+            try {
+                groupReceiver.add(groupReceiverListFileKey);
+                checkGroupReceiverListFile(groupReceiverListFileKey);
+            } catch (Exception e) {
+                isError = true;
+                setStatusForException(e, "An error occurred when retrieving receiver list, please try again");
+            }     
+        }
+        
+        if(addressModeOn){
+            addressReceiver.add(addressReceiverListString);          
+            try {
+                checkAddressReceiverString(addressReceiverListString);
+            } catch (InvalidParametersException e) {
+                isError = true;
+                setStatusForException(e);
+            }  
+        } 
+        
+        if(!addressModeOn && !groupModeOn){
+            isError = true;
+            statusToAdmin = "Error : No reciver address or file given";
+            statusToUser.add("Error : No reciver address or file given");       
+        }
+        
+        if(isError){
+            data.emailToEdit = new AdminEmailAttributes(subject,
+                                                        addressReceiver,
+                                                        groupReceiver,
+                                                        new Text(emailContent),
+                                                        null);
+            return createShowPageResult(Const.ViewURIs.ADMIN_EMAIL, data);
+        }
+        
+        boolean isEmailDraft = emailId != null && !emailId.isEmpty();
+        
+
+        if(!isEmailDraft) {
+            recordNewSentEmail(subject, addressReceiver, groupReceiver, emailContent);
+        } else {
+            updateDraftEmailToSent(emailId, subject, addressReceiver, groupReceiver, emailContent);
+        }
+ 
+        if(isError){
+            data.emailToEdit = new AdminEmailAttributes(subject,
+                                                        addressReceiver,
+                                                        groupReceiver,
+                                                        new Text(emailContent),
+                                                        null);
+        }
+        return createShowPageResult(Const.ViewURIs.ADMIN_EMAIL, data);
+    }
+    
+    private void checkAddressReceiverString(String addressReceiverString) throws InvalidParametersException{
+       FieldValidator validator = new FieldValidator();
+       
+       String[] emails = addressReceiverString.split(",");
+       for(String email : emails){
+           String error = validator.getInvalidityInfo(FieldType.EMAIL, email);
+           if(error != null && !error.isEmpty()){
+               isError = true;
+               statusToUser.add(error);
+               throw new InvalidParametersException("<strong>Email Format Error</strong>");
+           }
+       }
+       
+    }
+    
+    private void checkGroupReceiverListFile(String listFileKey) throws IOException {
```

Comment:
Not immediately clear what this method does. To this method and other methods that are long, 1. try to simplify using SLAP. 2. add more comments.